app/main.py

```diff
@@ -1,2 +1,63 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def resize(self) -> None:
+        self.capacity *= 2
+        self.size = 0
+        old_table = self.hash_table
+        self.hash_table = [None] * self.capacity
+
+        for node in old_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = hash(key) % self.capacity
+
+        while True:
```

Comment:
In your current solution, if the table reaches full capacity without resizing, the while True loop could be in an infinite loop.